src/com/facebook/buck/android/MergeAndroidResourcesStep.java

```diff
@@ -216,29 +230,47 @@ public StepExecutionResult execute(ExecutionContext context)
               filesystem,
               useOldStyleableFormat);
 
+      ImmutableSet.Builder<String> requiredPackages = ImmutableSet.<String>builder();
+
+      // Create a temporary list to avoid concurrent modification problems.
+      ArrayList<Entry<String, RDotTxtEntry>> entries =
+          new ArrayList<>(rDotJavaPackageToResources.entries());
+
+      if (skipPrebuiltRDotJava) {
+        // If skip_prebuilt_r_dot_java is true remove all packages except union package
+        Preconditions.checkArgument(
```

Comment:
This should also be checked in the Description (or the rule constructor) to ensure that we catch this early.

Reply:
added a check in the rule constructor.